editor/animation_track_editor.cpp

```diff
@@ -5181,7 +5189,17 @@ void AnimationTrackEditor::_update_key_edit() {
 				_clear_key_edit();
 				return; // Probably in the process of rearranging the keys.
 			}
-			key_ofs_map[track].push_back(animation->track_get_key_time(track, E.key.key));
+
+			float ofs = animation->track_get_key_time(track, E.key.key);
+			key_ofs_map[track].push_back(ofs);
+
+			// If we are editing the keyframe under the play position, we make sure
+			// to update the timeline when the properties are changed, we use the boolean
+			// to bind this signal once
+			if (!timeline_changed_bound && Math::is_equal_approx(timeline->get_play_position(), ofs)) {
+				timeline_changed_bound = true;
+				multi_key_edit->connect(CoreStringNames::get_singleton()->property_list_changed, callable_mp(this, &AnimationTrackEditor::_timeline_changed).bind(ofs, false, false));
+			}
```

Comment:
It looks redundant. You can rearrange the code by putting the `timeline_changed` flag outside of the if state, and only updating it once after the if state.